build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/RestTestBasePlugin.java

```diff
@@ -48,15 +61,22 @@ public void apply(Project project) {
                 runnerNonInputProperties.systemProperty(TESTS_CLUSTER, () -> String.join(",", cluster.getAllTransportPortURI()));
                 runnerNonInputProperties.systemProperty(TESTS_CLUSTER_NAME, cluster::getName);
             } else {
-                if (System.getProperty(TESTS_CLUSTER) == null || System.getProperty(TESTS_CLUSTER_NAME) == null) {
+                if (systemProperty(TESTS_CLUSTER) == null || systemProperty(TESTS_CLUSTER_NAME) == null) {
                     throw new IllegalArgumentException(
                         String.format("%s, %s, and %s must all be null or non-null", TESTS_REST_CLUSTER, TESTS_CLUSTER, TESTS_CLUSTER_NAME)
                     );
                 }
             }
         });
+
+        project.getTasks().named(JavaBasePlugin.CHECK_TASK_NAME).configure(check -> check.dependsOn(project.getTasks().withType(RestIntegTestTask.class)));
```

Comment:
I recall there's a _reason_ we don't do this, which is that it'll mean running `check` on BWC qa projects will run the tests for _every_ version, instead of just snapshots, which is [what we want](https://github.com/elastic/elasticsearch/blob/d8e4e48a3bccc8de8e1f0740294929cfa893f987/build-tools-internal/src/main/groovy/elasticsearch.bwc-test.gradle#L36).

Something like `./gradlew :qa:full-cluster-restart:check` would be a good test

Reply:
I see. Will move the wiring of the yamlRestTest and javaRestTest task back to the according plugins.